holmes/plugins/toolsets/investigator/core_investigation.py

```diff
@@ -0,0 +1,159 @@
+import logging
+import os
+from typing import Any, Dict
+
+from uuid import uuid4
+from holmes.core.todo_manager import (
+    get_todo_manager,
+)
+
+from holmes.core.tools import (
+    Toolset,
+    ToolsetTag,
+    ToolsetStatusEnum,
+    ToolParameter,
+    Tool,
+    StructuredToolResult,
+    ToolResultStatus,
+)
+from holmes.plugins.toolsets.investigator.model import Task, TaskStatus
+
+
+class TodoWriteTool(Tool):
+    name: str = "TodoWrite"
+    description: str = "Save investigation tasks to break down complex problems into manageable sub-tasks. ALWAYS provide the COMPLETE list of all tasks, not just the ones being updated."
```

Comment:
The list of calls should have no dependencies, so it's ok to run them in parallel.